extensions/roc-plugin-repo/src/commands/release.js

```diff
@@ -170,85 +192,190 @@ export default projects => ({
-                        fs.readFileSync(
-                          path.resolve(status[project].path, 'package.json'),
-                        ),
-                      ).version;
-                      // eslint-disable-next-line no-param-reassign
-                      status[project].version = newVersion;
-                      return execute(
-                        `git add . && git commit -m "release(${project}): ${newVersion}"`,
+                    previous.then(() =>
+                      execute(
+                        `git add . && git commit -m "release(${project.name}): ${project
+                          .packageJSON.version}"`,
                         {
                           silent: true,
-                          cwd: status[project].path,
+                          cwd: project.path,
                         },
-                      );
-                    }),
+                      ).then(async () => {
+                        const {
+                          stdout,
+                        } = await require('./utils/execute').default(
+                          'git rev-parse HEAD',
+                        );
+                        const hash = stdout.trim();
+                        // eslint-disable-next-line no-param-reassign
+                        project.releaseCommitHash = hash;
+                      }),
+                    ),
                   Promise.resolve(),
                 ),
             },
+            {
+              title: 'Creating tags',
+              skip: () => !git || !tag,
+              task: async ctx => {
+                // Always create individual tags for each package on release
+                if (isMonorepo) {
+                  return Promise.all(
+                    selectedToBeReleased.map(project => {
+                      // eslint-disable-next-line no-param-reassign
+                      project.tag = `${project.name}@${project.packageJSON
+                        .version}`;
+                      return execute(
+                        `git tag ${project.name}@${project.packageJSON
+                          .version} ${project.releaseCommitHash}`,
+                        { silent: true },
+                      );
+                    }),
+                  ).then(async () => {
+                    if (collectedRelease) {
+                      const releaseTag = await getTag(collectedRelease);
+                      ctx.releaseTag = releaseTag;
+                      return execute(`git tag ${releaseTag}`, { silent: true });
+                    }
+
+                    return Promise.resolve();
+                  });
+                }
+
+                selectedToBeReleased[0].tag = `v${selectedToBeReleased[0]
+                  .packageJSON.version}`;
+                return execute(
+                  `git tag v${selectedToBeReleased[0].packageJSON.version}`,
+                  {
+                    silent: true,
+                  },
```

Comment:
Same with first argument passed to execute - sometimes it is newline and sometimes it is in the same line. Not a big deal but following previous commands It should be like this:
```
execute(`git tag v${selectedToBeReleased[0].packageJSON.version}`, {
  silent: true,
},
```

I think execute command is pretty crucial for this part of the project and formatting should be clear for better understanding.